source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -282,6 +282,24 @@ std::string SPIFFEValidator::extractTrustDomain(const std::string& san) {
   return "";
 }
 
+absl::optional<Ssl::CertificateNameExpiration>
+SPIFFEValidator::getSecondsUntilExpirationOfCert() const {
+  // TODO(yleng): With the current interface, we cannot pass the multiple cert information.
```

Comment:
I think it would be more correct to return the value for the cert that expires earliest

Reply:
Updated to return a vector based on below comment: https://github.com/envoyproxy/envoy/pull/27687#discussion_r1243928956